src/main.rs

```diff
@@ -104,10 +105,19 @@ async fn initialize_astarte_device_sdk(
         return Err(AstarteMessageHubError::MissingConfig("interface directory"));
     };
 
+    let int_dir = int_dir
+        .as_os_str()
+        .to_str()
+        .ok_or(AstarteMessageHubError::MissingConfig(
+            "invalid interface directory",
+        ))?;
+
     // create a device instance
     let (device, rx_events) = DeviceBuilder::new()
         .interface_directory(int_dir)?
         .store(MemoryStore::new())
+        // the connect method will internally create a MessageHubClient already configured with an NodeId Interceptor
+        // layer, necessary to insert the Node ID into the gRPC metadata
```

Comment:
I don't understand this comment. It makes it seem like it is creating a Grpc connection with a MessageHub client, while in reality, this is just the connection to Astarte, if I'm not mistaken.

Reply:
It was an old comment. I've removed it